pandas/formats/format.py

```diff
@@ -2138,6 +2016,31 @@ def formatter(value):
 
         return formatter
 
+    def _get_formatted_data(self):
+        """
+        Returns formatted float values. This is different from
```

Comment:
The point you refer to was well-intended but ultimately incorrect, and sure, we could add this to `get_result_as_array`, but that's an API change then.